autosubmit/git/autosubmit_git.py

```diff
@@ -301,3 +303,19 @@ def check_unpushed_changes(expid: str) -> None:
                         # M: Modified, A: Added, D: Deleted, ?: Untracked
                         raise AutosubmitCritical("Push local changes to remote repository before running", 7075)
 
+    @staticmethod
+    def check_directory_in_use(expid: str) -> bool:
+        """
+        Checks for open files and unfnished git-credential requests in a directory
+        """
+        if process_id(expid) is not None:
+            return True
+        for proc in psutil.process_iter(['pid', 'name', 'cmdline']):
+            name = proc.info.get('name', '')
+            cmdline = proc.info.get('cmdline') or []
+            if 'git-credential' in name or any('git-credential' in arg for arg in cmdline):
+                return True
+        if bool(psutil.Process().open_files()):
+            return True
```

Comment:
I think this `if bool()` means: If the current process has any open files, then abort.

But the current process is the `autosubmit delete` command, and we shouldn't abort if it has any files (e.g. libs?) open, I think? Only if something like `autosubmit run` is running?